lib/tslint/5.12.x/tslint.json

```diff
@@ -0,0 +1,188 @@
+    "prefer-conditional-expression": false,
+    "prefer-object-spread": true,
+    "prefer-template": false,
+    "radix": true,
+    "switch-default": true,
+    "triple-equals": true,
+    "use-isnan": true,
+    "cyclomatic-complexity": [true, 20],
+    "no-default-export": true,
+    "no-duplicate-imports": true,
+    "no-mergeable-namespace": true,
+    "no-require-imports": false,
+    "object-literal-sort-keys": false,
+    "align": [false, "arguments", "statements", "parameters", "members", "elements"],
+    "callable-types": true,
+    "class-name": true,
+    "comment-format": [true, "check-space"],
+    "file-header": false,
+    "encoding": false,
+    "interface-name": [false, "never-prefix"],
+    "interface-over-type-literal": true,
+    "jsdoc-format": [true, "check-multiline-start"],
+    "no-angle-bracket-type-assertion": false,
+    "no-parameter-properties": false,
+    "no-redundant-jsdoc": true,
+    "no-reference-import": true,
+    "no-unnecessary-callback-wrapper": false,
+    "object-literal-shorthand": false,
+    "one-variable-per-declaration": true,
+    "switch-final-break": false,
+    "variable-name": [
+      true,
+      "check-format",
+      "require-const-for-all-caps",
+      "allow-leading-underscore",
+      "allow-trailing-underscore",
+      "ban-keywords"
+    ],
+    "whitespace": [
+      false,
+      "check-branch",
+      "check-decl",
+      "check-operator",
+      "check-module",
+      "check-separator",
+      "check-rest-spread",
+      "check-type",
+      "check-typecast",
+      "check-type-operator",
+      "check-preblock"
+    ],
+    "prefer-const": [
+      true,
+      {
+        "destructuring": "all"
+      }
+    ],
+    "no-magic-numbers": false,
+    "import-blacklist": [true, "@angular/material", "lodash", "rxjs/internal", "rxjs/index", "rxjs/Rx", "lodash-es"],
+    "ordered-imports": false,
+    "max-classes-per-file": false, // [true, 1],
+    "prefer-for-of": true,
+    "array-type": false, // [true, "array"]
+    "arrow-return-shorthand": true,
+    "binary-expression-operand-order": false,
+    "no-unnecessary-initializer": true,
+    "no-misused-new": true,
+    "prefer-method-signature": false,
+    "prefer-switch": [
+      true,
+      {
+        "min-cases": 3
+      }
+    ],
+    "prefer-function-over-method": false,
+    "prefer-while": true,
+    "no-import-side-effect": [
+      true,
+      {
+        "ignore-module": "(reflect-metadata|core-js|ngrx-store|angular|moment|polyfills|vendor|typings|\\.css|\\.pcss)"
+      }
+    ],
+    "no-non-null-assertion": true,
+    "no-parameter-reassignment": false,
+    "adjacent-overload-signatures": true,
+    "ban-comma-operator": false,
+    "ban-types": false,
+    "no-duplicate-super": true,
+    "no-duplicate-switch-case": true,
+    "newline-before-return": false,
+    "newline-per-chained-call": false,
+    "file-name-casing": [true, { "public_api.ts": "snake-case", ".*": "kebab-case" }],
```

Comment:
In fact I didn't add it to TSLint 5.11.x because the `snake-case` option of that rule was only added until TSLint 5.12, and if we would enable this rule already in TSLint 5.11 without the `snake-case` option then we would get linting errors due to the `public_api.ts` files that we have in Stark.

Do you think we should enable it anyway in TSLint 5.11 anyway?

Reply:
You're right! Let's keep it in 5.12 👍